src/symmetric_crypto/aes_256_gcm_pure/mod.rs

```diff
@@ -3,27 +3,29 @@
 //!
 //! It will use the AES native interface on the CPU if available.
 use crate::{
-    symmetric_crypto::{nonce::NonceTrait, SymKey, SymmetricCrypto},
+    symmetric_crypto::{nonce::NonceTrait, SymmetricCrypto},
+    typenum::U32,
     CryptoCoreError,
 };
+use aes::cipher::Unsigned;
 use aes_gcm::{
     aead::{generic_array::GenericArray, Aead, NewAead, Payload},
     AeadInPlace, Aes256Gcm,
 };
-use std::{fmt::Display, vec::Vec};
+use std::fmt::Display;
 
 pub mod dem;
 
 /// Use a 256-bit AES key
-pub const KEY_LENGTH: usize = 32;
+pub type KeyLength = U32;
```

Comment:
use full identifier `typenum::U32` here, because it is too easy to misread with `u32`

Reply:
ok